### DISCUSSION TITLE
Passing secret variable to Terraform/openTofu task not working

### DISCUSSION THREAD

**Discussion by fm-knopfler** [semaphoreui/semaphore#2287] [Q&A]
I set the following survey variable as a secret.
![grafik](https://github.com/user-attachments/assets/97087742-a76b-41f3-be04-024df4acb0f2)

And declared it like this in my terraform code.
```terraform
variable "api_token" {
  type      = string
  sensitive = true
}
```

Still, I get this error. With pure string variables (like hostname) it works fine. But as soon as I define it as a secret in either Semaphore or my TF code (or both), I get this error.
```
3:31:08 PM If you ever set or change modules or backend configuration for OpenTofu,
3:31:08 PM rerun this command to reinitialize your working directory. If you forget, other
3:31:08 PM commands will detect it and remind you to do so if necessary.
3:31:08 PM var.api_token
3:31:08 PM╷
3:31:08 PM│ Error: No value for required variable
3:31:08 PM│ 
3:31:08 PM│   on variables.tf line 11:
3:31:08 PM│   11: variable "api_token" {
3:31:08 PM│ 
3:31:08 PM│ The root module input variable "api_token" is not set, and has no default
3:31:08 PM│ value. Use a -var or -var-file command line argument to provide a value for
3:31:08 PM│ this variable.
```

Am I doing something wrong or is this a bug?

**Comment by ejfattouch** [comment]
I'm having the same issue, I believe its a bug

**Comment by jriberg** [comment]
I got a problem with this too.
I'm running Tofu with http backend. It works when I manually  pull the repo and run "tofu init" with TF_HTTP_USERNAME and TF_HTTP_PASSWORD as environmentvariables. 
I've added TF_HTTP_USERNAME and TF_HTTP_PASSWORD to my environment in semaphoreui, but I still get that the http backend requires auth, same error I get if the environment variables arent set

**Comment by rGunti** [comment]
Have a similar issue here. I have an environment configured with AWS credentials. When running the Tofu module, it complains about missing credentials. However, when I create a Bash script and run that to print the whole ENV, everything shows up as expected. I even called the tofu executable from my Bash script and it ran as expected.

Edit: Updating to v2.10.35 (was on .30) fixed the issue.
